### PR TITLE
Services - move waiter to package

### PR DESCRIPTION
**Pull request by hcsa73**

### CODE REVIEW

File: CONTRIBUTION.md [hunk 1]
```diff
@@ -8,7 +8,7 @@ Your contribution is welcome! Thank you for your interest in contributing to the
 
 ## Developer Guide
 ### Repository structure
-The SDK STACKIT service modules are located under `services`, which are automatically generated from the [REST API specs](https://github.com/stackitcloud/stackit-api-specifications), except for the `wait.go` and `wait_test.go` files. Therefore, changes to these modules besides the `wait.go` and `wait_test.go` files will not be accepted. Instead, consider proposing changes to the generation process in the [Generator repository](https://github.com/stackitcloud/stackit-sdk-generator).
+The SDK STACKIT service modules are located under `services`. The files located in `services/[service]` are are automatically generated from the [REST API specs](https://github.com/stackitcloud/stackit-api-specifications), whereas the ones located in subfolders (like `wait`) are manually maintained. Therefore, changes to files located in `services/[service]` will not be accepted. Instead, consider proposing changes to the generation process in the [Generator repository](https://github.com/stackitcloud/stackit-sdk-generator).
```
Comment (vicentepinto98) [inline CONTRIBUTION.md:11]:
```suggestion
The SDK STACKIT service modules are located under `services`. The files located in `services/[service]` are automatically generated from the [REST API specs](https://github.com/stackitcloud/stackit-api-specifications), whereas the ones located in subfolders (like `wait`) are manually maintained. Therefore, changes to files located in `services/[service]` will not be accepted. Instead, consider proposing changes to the generation process in the [Generator repository](https://github.com/stackitcloud/stackit-sdk-generator).
```

File: examples/waiter/waiter.go [hunk 2]
```diff
@@ -8,6 +8,7 @@ import (
 
 	"github.com/stackitcloud/stackit-sdk-go/core/utils"
 	"github.com/stackitcloud/stackit-sdk-go/services/dns"
+	"github.com/stackitcloud/stackit-sdk-go/services/dns/wait"
```
Comment (vicentepinto98) [inline examples/waiter/waiter.go:11]: I would go with the alias `dnsWait` to make it more clear, since we also have the `core/wait` pkg

**Review by vicentepinto98 — APPROVED** [review]

File: examples/waiter/waiter.go (hunk 2 above)
Comment (hcsa73) [reply examples/waiter/waiter.go:11]: I'd leave it like this. It's short and clear, looking at the code you'll easily assume that it's the wait for the dns, and it's unlikely conflict with other wait packages will happen: I don't think you ever want to import `core/wait`, and if you end up using waiters from multiple services, one can then use `[SERVICE]wait` for an alias

(Also small note, package names should be lower case be convention: https://go.dev/blog/package-names, so I'd keep the alias lower case as well)

Comment (hcsa73) [reply examples/waiter/waiter.go]: Had to remove waiter example, the lint was complaining that the dns release didn't have the `wait` package

**Review by vicentepinto98 — APPROVED** [review]

**Pull request merged**